### PR TITLE
Adobe Target VEC Documentation

### PR DESCRIPTION
**Pull request by kumarnimit**
Please note that the image used is from a this branch and may need to reference the master. The VEC extension is planned for release on Monday - 05/13/2019.

### CODE REVIEW

File: using-mobile-extensions/adobe-target-vec.md [hunk 1]
```diff
@@ -0,0 +1,144 @@
+# Adobe Target - VEC
+This extension enables Target VEC (Visual Experience Composer) functionality. The Target VEC for Native Mobile Apps lets you create activities and personalize content on native mobile apps in a do-it-yourself fashion without continuous development dependencies and app-release cycles.
```
Comment (rekrishn) [inline using-mobile-extensions/adobe-target-vec.md]: Please reword as follows:

"This extension enables the Adobe Target Visual Experience Composer (Target VEC)."

File: using-mobile-extensions/adobe-target-vec.md [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+
+{% hint style="info" %}
+Review the following Target VEC functionality documentation for context before implementation at these links:
```
Comment (rekrishn) [inline using-mobile-extensions/adobe-target-vec.md]: Please change to:

"For more information about Target VEC, see the following links in the Target VEC documentation:"

File: using-mobile-extensions/adobe-target-vec.md [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+
+* [Mobile App Visual Experience Composer](https://docs.adobe.com/content/help/en/target/using/implement-target/mobile-apps/composer/mobile-visual-experience-composer.html)
+* [Setup Mobile VEC for Android](https://docs.adobe.com/content/help/en/target/using/implement-target/mobile-apps/composer/mobile-visual-experience-composer-android.html)
+* [Setup Mobile VEC for iOS](https://docs.adobe.com/content/help/en/target/using/implement-target/mobile-apps/composer/mobile-visual-experience-composer-ios.html)
+* [Set up click tracking in the Mobile VEC](https://docs.adobe.com/content/help/en/target/using/implement-target/mobile-apps/composer/set-up-click-tracking-in-the-mobile-vec.html)
+{% endhint %}
+
+To get started with Target VEC, follow these steps:
```
Comment (rekrishn) [inline using-mobile-extensions/adobe-target-vec.md]: Please change to:

To get started, complete the following steps:

1. Configure the Target VEC extension.

  The The VEC extension is dependent on the Adobe Target Extension. Make sure Adobe Target extension is already configured and enabled.

2. Add the Target VEC Extension to your app. --> missing a period at the end.
3. Select one of the following implementation methods:

    * Auto-Fetch Target Activities, where the options are:

       * To fetch in a blocking call (background is OFF)
       * To fetch in an asynchronous mode (background is ON)

    * Fetch Target activities programmatically
    * Handle Target workspace restrictions

File: using-mobile-extensions/adobe-target-vec.md [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+
+1. Configure the Adobe Target - VEC extension. The VEC extension is dependent on the Adobe Target Extension. Make sure Adobe Target extension is already configured and enabled.
```
Comment (rekrishn) [inline using-mobile-extensions/adobe-target-vec.md]: Please change to:

To get started, complete the following steps:

Configure the Target VEC extension.

The The VEC extension is dependent on the Adobe Target Extension. Make sure Adobe Target extension is already configured and enabled.

Add the Target VEC Extension to your app. --> missing a period at the end.

Select one of the following implementation methods:

Auto-Fetch Target Activities, where the options are:

To fetch in a blocking call (background is OFF)
To fetch in an asynchronous mode (background is ON)
Fetch Target activities programmatically

Handle Target workspace restrictions

File: using-mobile-extensions/adobe-target-vec.md [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+   2. Fetch Target Activities Programmatically
+   3. Handle Target Workspace Restrictions
+
+## Configure the Adobe Target - VEC extension in Launch   <a id="configuring-the-adobe-target-vec-extension-in-adobe-launch"></a>
+
+![Adobe Target VEC Extension Configuration](../.gitbook/assets/adobe-target-vec-1.png)
+
+1. In Launch, click the **Extensions** tab.
+2. On the **Installed** tab, locate the Adobe Target VEC extension, and click **Configure**.
+3. The default configuration options loads Target VEC activities as a blocking call on App launch. See [Implementation Methods for Target VEC](implementation-methods-for-target-vec) for details.
```
Comment (rekrishn) [inline using-mobile-extensions/adobe-target-vec.md]: Please change the 2nd line to:

"For more information, see..."

File: using-mobile-extensions/adobe-target-vec.md [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+  ACPTarget.registerExtension()
+  
+  return true
+}
+```
+{% endtab %}
+{% endtabs %}
+
+## Implementation Methods for Target VEC   <a id="implementation-methods-for-target-vec"></a>
+The Target VEC extension retrieves the relevant Target experiences for your app through a single network request. Offers are retrieved via this network call and applied automatically on the targeted screens. It should be noted that no network requests are made to retrieve VEC experiences as the user navigates through multiple screens of the app.
```
Comment (rekrishn) [inline using-mobile-extensions/adobe-target-vec.md]: - please remove the word "single".

- Please remove "It should be noted" and start the sentence with "No network requests...".

File: using-mobile-extensions/adobe-target-vec.md [hunk 7]
```diff
@@ -0,0 +1,144 @@
...
+
+The default behavior of the extension is to make a synchronous network request (blocking call) at the time of Application launch. You can use Launch to control the behavior of this network request to meet your application behavior.
+
+### Auto-Fetch Target Activities
+This is the default behavior where a network request is initiated automatically by the Target VEC extension. You can use one of the following options to make this request a blocking call or an asynchronous request.
+
+#### Fetch in a synchronous call (background is OFF)
```
Comment (rekrishn) [inline using-mobile-extensions/adobe-target-vec.md]: as "Fetch in a synchronous call" is an option of "Auto-Fetch Target Activites", I'd recommend putting both of the "Fetch" options in a bulleted list like this:

### Auto-Fetch Target Activities

 * Fetch in a synchronous call (background is OFF)
* Fetch in an asynchronous call (background is ON)

File: using-mobile-extensions/adobe-target-vec.md [hunk 8]
```diff
@@ -0,0 +1,144 @@
...
+When selected Target VEC extension makes a network request as a blocking call on App launch. Offers are applied immediately and there is no flicker in the app. This is the default behavior of the extension.
```
Comment (rekrishn) [inline using-mobile-extensions/adobe-target-vec.md]: please add a comma between "selected" and "Target VEC extension"

File: using-mobile-extensions/adobe-target-vec.md [hunk 9]
```diff
@@ -0,0 +1,144 @@
...
+
+#### Fetch in an asynchronous call (background is ON)
+When selected Target VEC extension makes a network request in the background on App launch, but does not block the app from loading. If your experiences are authored on the home screen of your app, the offers may not apply for the home screen if the screen is rendered (this is typically when the lifecycle events `didFinishLaunchingWithOptions` and `onActivityResumed`  are triggered in iOS and Android respectively) before the call completes. Offers will be automatically apply on all subsequent screens.
```
Comment (rekrishn) [inline using-mobile-extensions/adobe-target-vec.md]: please add a comma between "selected" and "Target VEC extension"

File: using-mobile-extensions/adobe-target-vec.md [hunk 10]
```diff
@@ -0,0 +1,144 @@
...
+
+### Fetch Target Activities Programmatically
+You can disable the Target VEC extension to make the network request automatically and decide to programmatically call the Extension API. This gives your developers control on how they want to integrate Target VEC offers in the App. The Target VEC extension has two static methods `prefetchOffers` and `prefetchOffersBackground` that can used to programmatically retrieve Target VEC offers.
+1. `prefetchOffers` method will hide the current screen until Target VEC offers are fetched. The offers are automatically applied to the current screen if applicable and the screen is visible again.
```
Comment (rekrishn) [inline using-mobile-extensions/adobe-target-vec.md]: the numbered list doesn't seem like a procedure. If it's not, please change to bullets and move under the heading.

File: using-mobile-extensions/adobe-target-vec.md (hunk 9 above)
Comment (kumarnimit) [reply using-mobile-extensions/adobe-target-vec.md]: It changes the meaning. I have edit it to this:
When selected, Target VEC extension makes a network request in the background on App launch but does not block the app from loading. If your experiences are authored on the home screen of your app, the offers may not apply for the home screen if the screen is rendered before the call completes. The app screen rendering is typically identified via the lifecycle events `didFinishLaunchingWithOptions` and `onActivityResumed`  on iOS and Android respectively. Offers will be automatically apply on all subsequent screens.

**Pull request merged**